cpp/src/aztec/honk/circuit_constructors/circuit_constructor_base.hpp

```diff
@@ -264,24 +270,24 @@ template <size_t program_width_> class CircuitConstructorBase {
      * */
     virtual void set_public_input(const uint32_t witness_index)
     {
-        bool does_not_exist = true;
-        for (size_t i = 0; i < public_inputs.size(); ++i) {
-            does_not_exist = does_not_exist && (public_inputs[i] != witness_index);
-        }
-        if (does_not_exist) {
-            public_inputs.emplace_back(witness_index);
-        }
-        ASSERT(does_not_exist);
```

Comment:
Not sure if this `ASSERT` is useful, since if it is triggered, the remaining composer failure is not triggered.